### ISSUE TITLE
Support multi-line string literals for traditional/interpreted strings

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#8280]
Should it be possible to split a traditional string literal across multiple lines?  Currently the examples below are not allowed.  

```
var s1 = "This string
          spans multiple
          lines";
```

```
var s2 = 'This string
          spans multiple
          lines';
```

Since these are interpreted strings (e.g. they translate '\n' into a single newline character) should they automatically strip leading whitespace?  This would make it possible to line up the leading characters on each line in the source code as the above example does, but remove the padding from the actual string.  Then `writeln(s1);` would print:

```
This string
spans multiple
lines
```

Currently, you can split a string across multiple lines by adding a backslash to the end of each line to escape the newline character, but this requires modifying each line in the string (except the last):

```
var s3 = 'This string\
          spans multiple\
          lines';
```

Prints as:

```
This string
          spans multiple
          lines
```

**Comment by mppf** [comment]
As far as I know, the only reason for Chapel not to allow such multi-line string literals is if it significantly helps with error messages from the compiler. I don't think it does (but somebody could certainly try to convince me otherwise). I think we inherited the rule from C, where (as I understand) it exists to simplify the tokenization / early compilation steps.

So in other words I think we should allow multi-line `"` string literals.

I'd really like for the only difference between `"""` strings and `"` strings to be whether or not the escaping inside of the string applies. For the question of stripping leading whitespace, I think #8281 is a good solution that can reasonably apply to any string type.

**Comment by ronawho** [comment]
I really don't like the idea of supporting multi-line single quote strings. I don't know of any language that does that, though maybe that's just lack of knowledge on my part.

What's a real use case for this?

**Comment by mppf** [comment]
Same use cases as `"""` except you might want to enable quoting behavior.

**Comment by ronawho** [comment]
>Same use cases as """ except you might want to enable quoting behavior.

What's a real example though?

Multi-line strings are convenient for pasting code in, or for SQL/XML stuff, but I can't think of a common case where you would want interpreted multi-line strings.

**Comment by daviditen** [comment]
> I don't know of any language that does that, though maybe that's just lack of knowledge on my part.

As one example, Julia has them.

```
julia> print("Hello
       world!")
Hello
world!
```

In both Python and Julia triple-quoted strings are interpreted.

**Comment by ben-albrecht** [comment]
> I'd really like for the only difference between """ strings and " strings to be whether or not the escaping inside of the string applies.

One advantage of this approach is that it would allow users to specify multi-line interpreted strings, whereas they currently cannot. If we decide against this, it may be worth revisiting that design decision.

**Comment by dmk42** [comment]
What if we used a solution like #8281 for this?  Then we could have just one kind of multiline string, the uninterpreted kind, but still get the functionality of an interpreted string.  Something like

```
"""
Some
    Multi\nline
string
""".interpret()
```

**Comment by arezaii** [comment]
During module review of strings and bytes, we decided this issue should have been closed with the addition of the `dedent()` method included in #15664
